llvm/lib/Target/AMDGPU/AMDGPUCallLowering.cpp

```diff
@@ -520,6 +520,16 @@ bool AMDGPUCallLowering::lowerFormalArgumentsKernel(
 
   // TODO: Align down to dword alignment and extract bits for extending loads.
   for (auto &Arg : F.args()) {
+    // Hidden arguments that are in the kernel signature must be preloaded to
+    // user SGPRs. Print a diagnostic error if a hidden argument is in the
+    // argument list and is not preloaded.
+    if (Arg.hasAttribute("amdgpu-hidden-argument")) {
+      DiagnosticInfoUnsupported NonPreloadHiddenArg(
+          *Arg.getParent(),
+          "hidden argument in kernel signature was not preloaded");
+      F.getContext().diagnose(NonPreloadHiddenArg);
```

Comment:
return false and comment (with debug print) this is just because it's not implemented in globalisel